### PR TITLE
chore: update demo hosting and dist changes

### PR DESCRIPTION
**Pull request by veegandhi**
## Description

- Rename `docs` to `www`, so we can use `docs` folder to serve document distribution build on github pages.
- Create a new `webpack.config.prod` to generate the `docs` distribution
- Fixes issue which was serving `svg` and `png` on github pages from right path due to new webpack builds
## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Locally and on CI.

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [x] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [x] I've two LGTMs/Approvals.
- [x] I've fixed or replied to all my code-review comments.
- [x] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: config/webpack.config.prod.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        removeEmptyAttributes: true,
+        removeStyleLinkTypeAttributes: true,
+        keepClosingSlash: true,
+        minifyJS: true,
+        minifyCSS: true,
+        minifyURLs: true,
+      },
+    }),
+    new webpack.DefinePlugin({
+      'process.env.NODE_ENV': '"production"',
```
Comment (bchew) [inline config/webpack.config.prod.js]: Is that many quotes required?

Comment (veegandhi) [reply config/webpack.config.prod.js]: I think it needs a stringified version

File: config/webpack.config.prod.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+      amd: 'moment',
+    },
+  },
+  module: {
+    rules: [
+      {
+        enforce: 'pre', // Lint before babel transpiles; fail fast on syntax
+        test: /\.(js|jsx)$/,
+        exclude: [/node_modules/, /(\.spec)\.(js|jsx)$/],
+        include: [paths.appSrc, paths.appDemo],
```
Comment (knilink) [inline config/webpack.config.prod.js]: Is this copied from existing code?
I think either `exclude: [/node_modules/]` or `include: [paths.appSrc, paths.appDemo]` should be enought, assuming specs are not under the dependency tree entry for the entry.

File: package.json [hunk 3]
```diff
@@ -10,16 +10,17 @@
   "license": "MIT",
   "scripts": {
     "protect-master": "./scripts/protect-master",
-    "clean": "rimraf dist/*",
+    "clean": "rimraf dist/* && rimraf docs/*",
```
Comment (knilink) [inline package.json]: `rimraf dist/* docs/*` should work

File: config/webpack.config.prod.js (hunk 2 above)
Comment (veegandhi) [reply config/webpack.config.prod.js]: @knilink - remove redundant `exclude` from all config's

**Comment by vinteo** [comment]
Merging...

**Pull request merged**